src/messages.cpp

```diff
@@ -910,16 +910,15 @@ namespace quicr::messages {
         buffer << UintVar(msg.group_id);
         buffer << UintVar(msg.object_id);
         buffer.push_back(msg.priority);
+        PushExtensions(buffer, msg.extensions);
+
         buffer << UintVar(msg.payload.size());
         if (msg.payload.empty()) {
-            // empty payload needs a object status to be set
             buffer << UintVar(static_cast<uint8_t>(msg.object_status));
-            PushExtensions(buffer, msg.extensions);
-        } else {
-            PushExtensions(buffer, msg.extensions);
-            buffer << msg.payload;
         }
+
         buffer << msg.payload;
```

Comment:
I suppose it's fine to call to encode payload when it's zero in size but we could have returned from the payload empty check block to prevent an attempt on zero size payload.